django/db/models/expressions.py

```diff
@@ -915,9 +915,12 @@ def get_source_expressions(self):
         return [self.expression]
 
     def get_group_by_cols(self, alias=None):
-        expression = self.expression.copy()
-        expression.output_field = self.output_field
-        return expression.get_group_by_cols(alias=alias)
+        if isinstance(self.expression, Expression):
```

Comment:
I'm not sure about using `Expression` here (see similar doubts in #13165) :thinking: I would rather use `WhereNode`, e.g. 
```suggestion
        if not isinstance(self.expression, WhereNode):
```
\cc @charettes